rainier-core/src/main/scala/com/stripe/rainier/sampler/LeapFrog.scala

```diff
@@ -122,47 +117,76 @@ private[sampler] case class LeapFrog(nVars: Int,
 }
 
 private[sampler] object LeapFrog {
-  def apply(context: Context): LeapFrog = {
-    val qs = context.variables
-    val ps = qs.map { _ =>
-      new Variable
-    }
-    val potential = new Variable
-    val stepSize = new Variable
-    val inputs: List[Variable] = (ps ++ qs) :+ potential :+ stepSize
-
-    val newPotential = context.density * -1
-    val grad = context.gradient.map { g =>
-      g * -1
+  def apply(density: DensityFunction): LeapFrog = {
+
+    val nVars = density.nVars
+    val potentialIndex = nVars * 2
+    val stepSizeIndex = potentialIndex + 1
+
+    def halfPs(inputs: Array[Double]): Unit = {
```

Comment:
could these all just be instance methods on LeapFrog now instead of it having to take them as params? (it would take `density` instead)

Reply:
Yep! Updated.